packages/daf-cli/src/server.ts

```diff
@@ -27,150 +27,162 @@ program
     const agent = getAgent(program.config)
     const { server: options } = createObjects(getConfig(program.config), { server: '/server' })
 
-    passport.use(
-      new Bearer.Strategy((token, done) => {
-        if (!options.apiKey || options.apiKey === token) {
-          done(null, {}, { scope: 'all' })
-        } else {
-          done(null, false)
-        }
-      }),
-    )
+    /**
+     * Ngrok configuration
+     */
+    let baseUrl = options.baseUrl
+    if (options.ngrok?.connect) {
+      baseUrl = await ngrok.connect({
+        addr: cmd.port || options.port,
+        subdomain: options.ngrok.subdomain,
+        region: options.ngrok.region,
+        authtoken: options.ngrok.authtoken,
+      })
+      app.set('trust proxy', 'loopback')
+    }
+    const hostname = parse(baseUrl).hostname
+
+    /**
+     *  Authentication setup
+     */
+    let authMiddleware = (req: any, res: any, next: any) => {
+      next()
+    }
+    let securityScheme
+    if (options.apiKey) {
+      console.log('🔐 Secured by API Key')
+      passport.use(
+        new Bearer.Strategy((token, done) => {
+          if (!options.apiKey || options.apiKey === token) {
+            done(null, {}, { scope: 'all' })
+          } else {
+            done(null, false)
+          }
+        }),
+      )
 
+      authMiddleware = passport.authenticate('bearer', { session: false })
+      securityScheme = 'bearer'
+    }
+
+    /**
+     * Exposing agent methods
+     */
     const exposedMethods = options.exposedMethods ? options.exposedMethods : agent.availableMethods()
+    const getAgentForRequest = async (req: express.Request) => agent
+
+    console.log('🗺  API base path', baseUrl + options.apiBasePath)
+    app.use(
+      options.apiBasePath,
+      authMiddleware,
+      AgentRouter({
+        getAgentForRequest,
+        exposedMethods,
+      }),
+    )
 
-    const apiBasePath = options.apiBasePath
+    /**
+     * Exposing OpenAPI schema
+     */
+    console.log('🗺  OpenAPI schema', baseUrl + options.schemaPath)
+    console.log('🧩 Available methods', agent.availableMethods().length)
+    console.log('🛠  Exposed methods', exposedMethods.length)
+    app.use(
+      options.schemaPath,
+      ApiSchemaRouter({
+        basePath: options.apiBasePath,
+        getAgentForRequest,
+        exposedMethods,
+        securityScheme,
+      }),
+    )
 
-    const agentRouter = AgentRouter({
-      basePath: apiBasePath,
-      getAgentForRequest: async (req) => agent,
-      exposedMethods,
-      serveSchema: false,
-    })
+    /**
+     * Handling 'did:web' requests ('/.well-known/did.json' and '/^\/(.+)\/did.json$/')
+     */
+    console.log('📋 DID Document ' + baseUrl + didDocEndpoint)
```

Comment:
`did:web` won't work unless it is served under a `https` url.
Here may be a good spot to warn users about this aspect.

Reply:
Added comment about this